### PR TITLE
Add vis instance segmentation func

### PR DESCRIPTION
**Pull request by knorth55**
Merge after #540 

- [x] add `vis_instance_segmentation`
- [x] add `test_vis_instance_segmentation_dataset.py`

```python
from chainercv.datasets import SBDInstanceSegmentationDataset
from chainercv.datasets import sbd_instance_segmentation_label_names
from chainercv.visualizations import vis_instance_segmentation
import matplotlib.pyplot as plt

dataset = SBDInstanceSegmentationDataset()
img, bbox, mask, label = dataset[0]
vis_instance_segmentation(
    img, bbox, mask, label, label_names=sbd_instance_segmentation_label_names,
    alpha=0.7)
plt.show()
```
ins_dataset_example

### CODE REVIEW

File: chainercv/visualizations/vis_instance_segmentation.py [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+        matploblib.axes.Axes: Returns :obj:`ax`.
+        :obj:`ax` is an :class:`matploblib.axes.Axes` with the plot.
+
+    """
+    if label is not None and not len(bbox) == len(label):
+        raise ValueError('The length of label must be same as that of bbox')
+    if score is not None and not len(bbox) == len(score):
+        raise ValueError('The length of score must be same as that of bbox')
+
+    n_ins_class = len(bbox)
```
Comment (yuyu2172) [inline chainercv/visualizations/vis_instance_segmentation.py]: s/n_ins_class/n_inst

`n_ins_class` sounds like this is related to class categories.

File: chainercv/visualizations/vis_instance_segmentation.py [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+    colors = [_default_cmap(l) for l in range(n_ins_class + 1)]
```
Comment (yuyu2172) [inline chainercv/visualizations/vis_instance_segmentation.py]: How about `range(1, n_inst)`?
By doing so, you can remove one line.
Also, `np.array` is optional in this case, so we can omit it.

File: chainercv/visualizations/vis_instance_segmentation.py [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+
+    Returns:
+        matploblib.axes.Axes: Returns :obj:`ax`.
+        :obj:`ax` is an :class:`matploblib.axes.Axes` with the plot.
+
+    """
+    if label is not None and not len(bbox) == len(label):
+        raise ValueError('The length of label must be same as that of bbox')
+    if score is not None and not len(bbox) == len(score):
+        raise ValueError('The length of score must be same as that of bbox')
```
Comment (yuyu2172) [inline chainercv/visualizations/vis_instance_segmentation.py]: How about checking `mask` as well?

File: chainercv/visualizations/vis_instance_segmentation.py [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+
+    n_ins_class = len(bbox)
+    colors = [_default_cmap(l) for l in range(n_ins_class + 1)]
+    colors = np.array(colors)[1:]
+
+    # Returns newly instantiated matplotlib.axes.Axes object if ax is None
+    ax = vis_image(img, ax=ax)
+
+    vis_img = np.zeros((mask.shape[1], mask.shape[2], 4), dtype=np.uint8)
```
Comment (yuyu2172) [inline chainercv/visualizations/vis_instance_segmentation.py]: `vis_img` has very similar name to `vis_image`.
How about `canvas`?

File: chainercv/visualizations/vis_instance_segmentation.py [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+        ...     label_names=sbd_instance_segmentation_label_names,
+        ...     alpha=0.7)
+        >>> plot.show()
+
+    Args:
+        img (~numpy.ndarray): An array of shape :math:`(3, H, W)`.
+            This is in RGB format and the range of its value is
+            :math:`[0, 255]`.
+        bbox (~numpy.ndarray): A float array of shape :math:`(R, 4)`.
+            R is the number of objects in the image, and each vector
```
Comment (yuyu2172) [inline chainercv/visualizations/vis_instance_segmentation.py]:
```
R --> `:math:`R``
```

File: chainercv/visualizations/vis_instance_segmentation.py [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+        img, bbox, mask, label=None, score=None, label_names=None,
+        alpha=1, ax=None):
+    """Visualize instance segmentation.
+
+    Example:
+
+        >>> from chainercv.datasets import SBDInstanceSegmentationDataset
+        >>> from chainercv.datasets \
+        ...     import sbd_instance_segmentation_label_names
+        >>> from chainercv.visualizations import vis_image
```
Comment (yuyu2172) [inline chainercv/visualizations/vis_instance_segmentation.py]: We do not need `vis_image`

**Comment by yuyu2172** [comment]
Please rebase master

**Comment by knorth55** [comment]
i rebased and updated

File: chainercv/visualizations/vis_instance_segmentation.py [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+        mask (~numpy.ndarray): A bool array of shape
+            :math`(R, H, W)`.
+            If there is an object, the value of the pixel is :obj:`True`,
+            and otherwise, it is :obj:`False`.
+        label (~numpy.ndarray): An integer array of shape :math:`(R, )`.
+            The values correspond to id for label names stored in
+            :obj:`label_names`.
+        label_names (iterable of strings): Name of labels ordered according
+            to label ids.
+        alpha (float): The value which determines transparency of the figure.
```
Comment (yuyu2172) [inline chainercv/visualizations/vis_instance_segmentation.py]: Is `alpha=1` a reasonable default value?
To me, `alpha=0.7` looks better.
Also, the doc needs to be changed if we are changing this.

Comment (knorth55) [reply chainercv/visualizations/vis_instance_segmentation.py]: I prefer alpha=0.7, so I will change it

Comment (knorth55) [reply chainercv/visualizations/vis_instance_segmentation.py]: I updated.

**Pull request merged**